product_supplierinfo_for_customer_sale/models/sale_order_line.py

```diff
@@ -9,28 +10,40 @@
 class SaleOrderLine(models.Model):
     _inherit = "sale.order.line"
 
+    customerinfo_id = fields.Many2one(
+        comodel_name="product.customerinfo", compute="_compute_customerinfo_id"
+    )
     product_customer_code = fields.Char(
         compute="_compute_product_customer_code",
     )
 
     @api.depends("product_id")
+    def _compute_customerinfo_id(self):
+        for line in self:
+            line.customerinfo_id = (
+                line.product_id._select_customerinfo(partner=line.order_partner_id)
+                if line.product_id
+                else line.customerinfo_id
+            )
+
+    @api.depends("customerinfo_id")
     def _compute_product_customer_code(self):
         for line in self:
-            if line.product_id:
-                supplierinfo = line.product_id._select_customerinfo(
-                    partner=line.order_partner_id
-                )
-                code = supplierinfo.product_code
-            else:
-                code = ""
-            line.product_customer_code = code
-
-    @api.onchange("product_id")
-    def product_id_change(self):
```

Comment:
Why is this not working?

Reply:
This onchange is not a problem, but if we have already set the `customerinfo_id` field I think it is better to make the onchange with that field to avoid redoing the search.

Reply:
The approach is not valid, as you are replacing any possible text, not exactly the reference we are looking for. Please revert to the previous condition.

Reply:
Changed to another approach (and explained why)